makefile

```diff
@@ -21,10 +65,17 @@ else
 	# Using separate stack-work directories to avoid recompiling when
 	# changing between debug and non-debug builds, per
 	# https://github.com/commercialhaskell/stack/issues/1132#issuecomment-386666166
-	PROF := --trace --work-dir .stack-work-prof
-
-	dex     := $(STACK) exec         dex --
-	dexprof := $(STACK) exec $(PROF) dex --
+	OPT  := --work-dir .stack-work-opt
+	DBG  := --work-dir .stack-work-dbg --trace
+	PROF := --work-dir .stack-work-prof --trace \
+	  --ghc-options="-fno-prof-auto" \
+          --ghc-options="-ddump-simpl" \
+          --ghc-options="-ddump-stg" \
+	  --ghc-options="-dsuppress-all" \
+          --ghc-options="-dno-suppress-type-signatures" \
+          --ghc-options="-ddump-to-file"
+
+	dex := $(STACK) exec dex --
```

Comment:
Should we perhaps delete the DEX_DUMP_GHC-based workflow?

Reply:
Done.